server.cc

```diff
@@ -2,38 +2,26 @@
 #include <string>
 #include <cstring>
 #include <stdexcept>
+#include <map>
 #include "server.h"
 #include "config_parser.h"
 #include "echo_handler.h"
 #include "file_handler.h"
+#include "not_found_handler.h"
 
 using boost::asio::ip::tcp;
 
 Server* Server::MakeServer(boost::asio::io_service& io_service, NginxConfig& out_config)
 {
+	// generate request handlers
+	auto handlers = new HandlerContainer();
```

Comment:
Server class will need a destructor to handle cleaning this.
Also why did we make this auto again?